### ISSUE TITLE
Working through an external proxy

### ISSUE THREAD

**Issue by misssis1** [sleeyax/burp-awesome-tls#46]
There is a need to set up an external proxy server (for ip spoofing). If you do it in Burp with the extension disabled and SOCKS5 proxy defined, everything works as it should.
![2](https://github.com/sleeyax/burp-awesome-tls/assets/151674518/2428dd8a-6d33-4cd3-86ef-89308253c34c)
If you enable the extension, the following error appears:
![1](https://github.com/sleeyax/burp-awesome-tls/assets/151674518/8a7f0e3d-9285-4897-8e08-f27f6e1caefb)
I take it the extension won't work through an external proxy?

**Comment by sleeyax** [comment]
Correct, this extension doesn't support external proxies yet. Probably should implement this after #38.

**Comment by L1ulee** [comment]
If SOCKS5 proxy is configured, a Network Unreachable error may also occur.

**Comment by sleeyax** [comment]
> If SOCKS5 proxy is configured, a Network Unreachable error may also occur.

Interesting, but please elaborate. Where did you configure the proxy? And could you show a screenshot of the error?

**Comment by L1ulee** [comment]
> > If SOCKS5 proxy is configured, a Network Unreachable error may also occur.
> 
> Interesting, but please elaborate. Where did you configure the proxy? And could you show a screenshot of the error?


I am unable to reproduce this issue, the environment is an Alibaba Cloud server, using Burpsuite -> Settings -> Network -> Connections -> Use SOCKS proxy -> Override options for this project only and configure SOCKS Proxy.Then using Firefox FoxyProxy to proxy 127.0.0.1:8080,and get this error. and when using another proxy server, there will also be issues, showing SOCKS: Connection Refused

<img width="560" alt="Image" src="https://github.com/user-attachments/assets/1740729c-fa0a-406c-8cfb-103c535cde9b" />

<img width="800" alt="Image" src="https://github.com/user-attachments/assets/42abe978-6847-4337-b411-79364ba06cb0" />

**Comment by shaneilahi** [comment]
hey how long is it going to take to add this feature?
